tests/unit/base/SGObject_unittest.cc

```diff
@@ -311,3 +312,87 @@ TEST(SGObject,parameter_hash_changed)
 	SG_UNREF(inf);
 }
 #endif //USE_GPL_SHOGUN
+
+TEST(SGObject, tags_set_float)
+{
+	Tag<float64_t> width("width");
+	CGaussianKernel* gauss_kernel = new CGaussianKernel();
+	EXPECT_EQ(gauss_kernel->set(width, 10.0), 10.0);
+	EXPECT_EQ(gauss_kernel->get(width), 10.0);
+	EXPECT_EQ(gauss_kernel->set("width", 30.0), 10.0);
+	EXPECT_EQ(gauss_kernel->get(width), 30.0);
+}
+
+TEST(SGObject, tags_set_sgvector)
+{
+	CDenseFeatures<int>* feats = new CDenseFeatures<int>();
+	SGVector<float64_t> x(3);
+	x[0]=0;
+	x[1]=1.1;
+	x[2]=2.2;
+	SGVector<float64_t> y = feats->set("training_data", x);
```

Comment:
this is not good, users should not be able to set/create arbritary tags for classes.
Rather they should only be able to modify existing tags. The public `set` should in fact throw an error if this tag is not existing in the map. There can be a protected `set` that can *add* new tags to the map, called from constructor just like the current `SG_ADD`
You can test that with a mock object on top of the unit test.